packages/commonwealth/client/scripts/views/pages/discussions/recent_threads_header.tsx

```diff
@@ -15,133 +13,124 @@ import { CWText } from '../../components/component_kit/cw_text';
 import { isWindowExtraSmall } from '../../components/component_kit/helpers';
 import { StagesMenu } from './stages_menu';
 import { TopicsMenu } from './topics_menu';
-import withRouter from 'navigation/helpers';
+import { useCommonNavigate } from 'navigation/helpers';
 
-type RecentThreadsHeaderAttrs = {
+type RecentThreadsHeaderProps = {
   stage: string;
   topic: string;
   totalThreadCount: number;
 };
 
-class RecentThreadsHeaderComponent extends ClassComponent<
-  RecentThreadsHeaderAttrs
-> {
-  private isWindowExtraSmall: boolean;
-
-  onResize() {
-    this.isWindowExtraSmall = isWindowExtraSmall(window.innerWidth);
-    redraw();
-  }
-
-  oninit() {
-    this.isWindowExtraSmall = isWindowExtraSmall(window.innerWidth);
-
-    window.addEventListener('resize', () => {
-      this.onResize();
-    });
-  }
-
-  onremove() {
-    window.removeEventListener('resize', () => {
-      this.onResize();
-    });
-  }
-
-  view(vnode: ResultNode<RecentThreadsHeaderAttrs>) {
-    const { topic, stage, totalThreadCount } = vnode.attrs;
-
-    const { stagesEnabled, customStages } = app.chain?.meta;
-
-    const topics = app.topics.getByCommunity(app.activeChainId());
-
-    const featuredTopics = topics
-      .filter((t) => t.featuredInSidebar)
-      .sort((a, b) => a.name.localeCompare(b.name))
-      .sort((a, b) => a.order - b.order);
-
-    const otherTopics = topics
-      .filter((t) => !t.featuredInSidebar)
-      .sort((a, b) => a.name.localeCompare(b.name));
-
-    const selectedTopic = topics.find((t) => topic && topic === t.name);
-
-    const stages = !customStages
-      ? [
-          ThreadStage.Discussion,
-          ThreadStage.ProposalInReview,
-          ThreadStage.Voting,
-          ThreadStage.Passed,
-          ThreadStage.Failed,
-        ]
-      : parseCustomStages(customStages);
-
-    const selectedStage = stages.find((s) => s === (stage as ThreadStage));
-
-    return (
-      <div className="RecentThreadsHeader">
-        {isUndefined(topic) && (
-          <>
-            <div className="header-row">
-              <CWText type="h3" fontWeight="semiBold" className="header-text">
-                All Discussions
+export const RecentThreadsHeader = (props: RecentThreadsHeaderProps) => {
+  const { topic, stage, totalThreadCount } = props;
```

Comment:
let's destructure props in the component param

Reply:
We might as well do all of these in one PR.

<img width="237" alt="Screenshot 2023-02-21 at 6 39 48 PM" src="https://user-images.githubusercontent.com/13058678/220431183-1deb4ebd-9081-48f5-a8a5-709f839de989.png">

Reply:
agree that we can do that all in one PR, but if we refactor some files, why can't we do it in a right way.